Lib/inspect.py

```diff
@@ -399,8 +399,6 @@ def _has_coroutine_mark(f):
     while ismethod(f):
         f = f.__func__
     f = functools._unwrap_partial(f)
-    if not (isfunction(f) or _signature_is_functionlike(f)):
```

Comment:
What else can we mark as coroutine-like? Functions, methods, classes. Anything else?

Reply:
Hi @sobolevn. I'm not sure I follow 🤔

The cases that have occurred up to Python 3.11, using the old `asyncio` version, have been callable classes and sync functions. Those are (now) covered. We also added tests for static and class method and lambdas (but I've not seen anyone using those).  

The other case that would simplify client code is automatic detection of the `__call__` method (either `async def` or marked) but I think the preference is not to add that.